### ISSUE TITLE
Not able to to generate "additionalProperties" property in swagger using cadl.

### ISSUE THREAD

**Issue by vishwahiremat** [microsoft/cadl#1473]
We have a swagger spec where we specify **"additionalProperties": true** in the model so user can add not defined properties as part of the model.
Example:
```
    "ModelProperties": {
      "description": "Model properties",
      "type": "object",
      "additionalProperties": true,   // additional properties flag
      "properties": {
        "provisioningState": {
          "readOnly": true,
          "$ref": "global.json#/definitions/ProvisioningState",
          "description": "Provisioning state  at the time the operation was called"
        }
      },
    },
```
And the generated client model code in go using autorest looks like below:
```
type ModelProperties struct {
	// OPTIONAL; Contains additional key/value pairs not defined in the schema.
	AdditionalProperties map[string]interface{}

	ProvisioningState *ProvisioningState `json:"provisioningState,omitempty" azure:"ro"`
}
```
with the default case in UnmarshalJSON funtion to add the not defined properties in AdditionalProperties map. i.e 
```
func (e *ModelProperties) UnmarshalJSON(data []byte) error {
	var rawMsg map[string]json.RawMessage
	if err := json.Unmarshal(data, &rawMsg); err != nil {
		return fmt.Errorf("unmarshalling type %T: %v", e, err)
	}
	for key, val := range rawMsg {
		var err error
		switch key {
		case "provisioningState":
				err = unpopulate(val, "ProvisioningState", &e.ProvisioningState)
				delete(rawMsg, key)
		default:
			if e.AdditionalProperties == nil {
				e.AdditionalProperties = map[string]interface{}{}
			}
			if val != nil {
				var aux interface{}
				err = json.Unmarshal(val, &aux)
				e.AdditionalProperties[key] = aux
			}
			delete(rawMsg, key)
		}
		if err != nil {
			return fmt.Errorf("unmarshalling type %T: %v", e, err)
		}
	}
	return nil
}
```
But i couldn't find a way to have a similar functionality using cadl. I have tried using **Record<unknown>**, when client code is generated I can see it create a map but not the default case in the UnmarshalJson.

I was hoping to have something like @AdditionalProperties annotation on the mode which would generate "additionalProperties" flag on the object in swagger, or is there any other way to achieve the same functionality?

[Playground Link](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=b3AgdGVzdCgpOiBGb287CgoKbW9kZWzEDSBpcyBSZWNvcmQ8dW5rbm93bj4gewogIHByb3Zpc2lvbmluZ1N0YXRlOiBzdHJpbmc7Cn0%3D)

**Comment by markcowl** [comment]
Related to #763 

You would actually want to require a constrained map, like string -> string for additional properties, I think, rather than just allowing additional properties of any type.

**Comment by markcowl** [comment]
Fix tracked in #1590
